R/checks.R

```diff
@@ -557,3 +558,48 @@ check_arg_is_valid_extension <- function(x,
 
   return(TRUE)
 }
+
+
+#' @title
+#' Check requested data arg exists in REDCap data
+#'
+#' @description
+#' Provide an error message when an argument is requested, but is not found in
+#' any \code{read_redcap()} `redcap_data` output.
+#'
+#' @details
+#' Currently used for the following arguments:
+##' \itemize{
+#'  \item{`export_survey_fields`: `*_timestamp`}
+#'  \item{`export_data_access_groups`: `redcap_data_access_group`}
+#' }
+#'
+#' @return
+#' An error message saying the requested data does not exist
+#'
+#' @importFrom cli cli_abort
+#' @importFrom rlang caller_env
+#'
+#' @param db_data The REDCap database output generated by
+#' \code{REDCapR::redcap_read_oneshot()$data}
+#' @param col The column to check for in `redcap_data`
+#' @param arg The argument used for the column check
+#' @param call The calling environment to use in the error message
+#'
+#' @keywords internal
+check_data_arg_exists <- function(db_data, col, arg, call = caller_env()) {
+
+  msg_x <- "Invalid request: {.arg {arg}}."
```

Comment:
I think this error message could be a bit more helpful. If you're trying to pull survey fields from a project that doesn't have any survey instruments, it should say so. And if you're trying to pull the DAG column from a non-DAG enabled project it should say so. Also, I wouldn't say "identifying column" because we use that term to describe the partial key columns.

Reply:
Sure, makes sense to me